Provide support for PHP 8.0

ittmann opened the pull request:
Adds support for PHP 8.0 as requested in #27

- composer.json already doesn't support PHP less than 7.3
- composer.json currently does not require-dev phpunit
(it is installed during initial project creation when MVC testing support is installed as described in README.md)
- no .travis.yml is present

Tested in docker with php:8.0-rc-apache
The skeleton app is compatible/runs without any issues.

phpunit reports an error on the use of the deprecated function libxml_disable_entity_loader in laminas-test, which should be fixed when PHP 8.0 support is added to laminas-test

```php
 composer run test
> phpunit
PHPUnit 9.4.2 by Sebastian Bergmann and contributors.

Warning:       Your XML configuration validates against a deprecated schema.
Suggestion:    Migrate your XML configuration using "--migrate-configuration"!

.E.                                                                 3 / 3 (100%)

Time: 00:00.028, Memory: 10.00 MB

There was 1 error:

1) ApplicationTest\Controller\IndexControllerTest::testIndexActionViewModelTemplateRenderedWithinLayout
Function libxml_disable_entity_loader() is deprecated

/var/www/vendor/laminas/laminas-dom/src/Document.php:234
/var/www/vendor/laminas/laminas-dom/src/Document.php:159
/var/www/vendor/laminas/laminas-dom/src/Document/Query.php:45
/var/www/vendor/laminas/laminas-test/src/PHPUnit/Controller/AbstractHttpControllerTestCase.php:386
/var/www/vendor/laminas/laminas-test/src/PHPUnit/Controller/AbstractHttpControllerTestCase.php:411
/var/www/vendor/laminas/laminas-test/src/PHPUnit/Controller/AbstractHttpControllerTestCase.php:434
/var/www/vendor/laminas/laminas-test/src/PHPUnit/Controller/AbstractHttpControllerTestCase.php:451
/var/www/module/Application/test/Controller/IndexControllerTest.php:48

ERRORS!
Tests: 3, Assertions: 6, Errors: 1.
Script phpunit handling the test event returned with error code 2
```

ittmann committed:
Provide support for PHP 8.0

Signed-off-by: Dick Ittmann <dick.ittmann@gmail.com>

froschdesign commented:
@ittmann 
> phpunit reports an error on the use of the deprecated function libxml_disable_entity_loader in laminas-test, which should be fixed when PHP 8.0 support is added to laminas-test

`libxml_disable_entity_loader` is used in laminas-dom and this [supports now PHP 8](https://github.com/laminas/laminas-dom/releases/tag/2.8.0).

samsonasik commented:
this may need laminas-skeleton-installer to support php 8 first, ref https://github.com/laminas/laminas-skeleton-installer/pull/19

samsonasik commented:
@ittmann please update `laminas/laminas-skeleton-installer` require to `^0.3` in composer.json. Thank you.

samsonasik commented on composer.json:
```diff
@@ -10,7 +10,7 @@
     "require": {
-        "php": "^7.3",
+        "php": "^7.3 || ~8.0.0",
         "laminas/laminas-component-installer": "^1.0 || ^2.1",
```
```suggestion
        "laminas/laminas-component-installer": "^3.0",
```

samsonasik replied:
the `laminas/laminas-component-installer` can be updated to use `^2.4`:

```suggestion
        "laminas/laminas-component-installer": "^2.4",
```

samsonasik commented on composer.json:
```diff
@@ -10,7 +10,7 @@
+        "php": "^7.3 || ~8.0.0",
         "laminas/laminas-component-installer": "^1.0 || ^2.1",
         "laminas/laminas-development-mode": "^3.2",
         "laminas/laminas-skeleton-installer": "^0.2 || ^1.0",
```
```suggestion
        "laminas/laminas-skeleton-installer": "^0.3",
```

samsonasik replied:
please update like above suggestion:

```diff
-        "laminas/laminas-skeleton-installer": "^0.2 || ^1.0",
+        "laminas/laminas-skeleton-installer": "^0.3",
```

ittmann replied:
Updated composer.json as suggested by @samsonasik

DIReports committed:
changes to required versions
"laminas/laminas-component-installer": "^2.4"
"laminas/laminas-skeleton-installer": "^0.3"

Signed-off-by: Dick Ittmann <dick.ittmann@reports.nl>

ittmann commented:
Tested again (in docker with php:8.0-apache) after the version changes suggested by samsonasik dc82d2c
composer install with minimal install option works fine and the application runs without any issues.

Choosing MVC testing support or installing laminas-test after initial project creation fails (PHP 8.0 support hasn't been added to laminas-test yet)

Everything works as expected with php:7.3-apache / the current project Dockerfile and docker-compose.yml

samsonasik reviewed (approved):
There are [extra components](https://github.com/ittmann/laminas-mvc-skeleton/blob/dc82d2c9afd0fbb1078937daa0ac5903c0c09497/composer.json#L29-L109) that need to be check against php 8,  but for basic installation, I think it is ready 👍

samsonasik commented:
I am merging it now, I will verify the latest components inside "extra" config.

samsonasik commented:
Thank you @ittmann
